a2/src/view/smart/SmartQuestionList.js

```diff
@@ -0,0 +1,40 @@
+import React, {Component} from "react"
+import questionListPresenter from "../../presenter/questionListPresenter"
+import questionModel from "../../model/questionModel"
+import QuestionList from "../dumb/QuestionList"
+
+const mapModelStateToComponentState = (modelState) => ({
+    questions: modelState.questions,
+    newQuestion: modelState.newQuestion,
+})
+
+export default class SmartQuestionList extends Component{
+    constructor(){
+        super();
+        this.state = mapModelStateToComponentState(questionModel.state);
+        this.state.questions.sort( (q1, q2) => (
```

Comment:
This actually changes the array in place (from the model). You should create a clone of the array and sort it inside the mapModelToBla function.